sdk/core/azure-core/src/azure_assert.cpp

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/core/azure_assert.hpp"
+
+void AzureNoReturnPath(std::string msg)
```

Comment:
We neither modify nor make a copy of `msg`, so it might as well be `const std::string&`. I'd also repeat the `[[noreturn]]` attribute here for consistency even though compilers don't care.